source/common/grpc/context_impl.cc

```diff
@@ -95,22 +114,16 @@ void ContextImpl::chargeResponseMessageStat(const Upstream::ClusterInfo& cluster
       .add(amount);
 }
 
-absl::optional<ContextImpl::RequestNames>
+absl::optional<ContextImpl::RequestStatNames>
 ContextImpl::resolveServiceAndMethod(const Http::HeaderEntry* path) {
```

Comment:
suggest renaming this method `resolveDynamicServiceAndMethod` to be clear this is going to do dynamic stat-name allocation.